src/pybind/mgr/balancer/module.py

```diff
@@ -388,6 +412,59 @@ def handle_command(self, inbuf, command):
                 self.active = False
             self.event.set()
             return (0, '', '')
+        elif command['prefix'] == 'balancer pool ls':
+            pool_ids = self.get_module_option('pool_ids')
+            if pool_ids is '':
+                return (0, '', '')
+            pool_ids = pool_ids.split(',')
+            pool_ids = [int(p) for p in pool_ids]
+            pool_name_by_id = dict((p['pool'], p['pool_name']) for p in self.get_osdmap().dump().get('pools', []))
+            should_prune = False
+            final_ids = []
+            final_names = []
+            for p in pool_ids:
+                if p in pool_name_by_id:
+                    final_ids.append(p)
+                    final_names.append(pool_name_by_id[p])
+                else:
+                    should_prune = True
+            if should_prune: # some pools were gone, prune
+                if final_ids:
+                    self.set_module_option('pool_ids', ','.join(final_ids))
+                else:
+                    self.set_module_option('pool_ids', '')
```

Comment:
`','.join(final_ids)` is already the empty string: 

```python
In [1]: ','.join([])
Out[1]: ''
```

Reply:
@sebastian-philipp Thanks!